pydoctor/templatewriter/writer.py

```diff
@@ -31,67 +38,86 @@ def __subclasshook__(cls, subclass: Type[object]) -> bool:
                     return False
         return True
 
-    def __init__(self, filebase):
-        self.base = filebase
-        self.written_pages = 0
-        self.total_pages = 0
-        self.dry_run = False
+    def __init__(self, filebase:str, template_lookup:Optional[TemplateLookup] = None):
+        """
+        @arg filebase: Output directory. 
+        @arg template_lookup: Custom L{TemplateLookup} object. 
+        """
+        self.base: Path = Path(filebase)
+        self.written_pages: int = 0
+        self.total_pages: int = 0
+        self.dry_run: bool = False
+        self.template_lookup:TemplateLookup = ( 
+            template_lookup if template_lookup else TemplateLookup() )
+        """Writer's L{TemplateLookup} object"""
 
-    def prepOutputDirectory(self):
+    def prepOutputDirectory(self) -> None:
+        """
+        Copy static CSS and JS files to build directory and warn when custom templates are outdated. 
+        """
         os.makedirs(self.base, exist_ok=True)
-        shutil.copyfile(templatefile('apidocs.css'),
-                        os.path.join(self.base, 'apidocs.css'))
-        shutil.copyfile(templatefile('bootstrap.min.css'),
-                        os.path.join(self.base, 'bootstrap.min.css'))
-        shutil.copyfile(templatefile('pydoctor.js'),
-                        os.path.join(self.base, 'pydoctor.js'))
+        shutil.copy(
+            self.template_lookup.get_template('apidocs.css').path,
+            self.base.joinpath('apidocs.css'))
+        shutil.copy(
+            self.template_lookup.get_template('extra.css').path,
+            self.base.joinpath('extra.css'))
+        shutil.copy(
+            self.template_lookup.get_template('bootstrap.min.css').path,
+            self.base.joinpath('bootstrap.min.css'))
+        shutil.copy(
+            self.template_lookup.get_template('pydoctor.js').path,
+            self.base.joinpath('pydoctor.js'))
```

Comment:
Instead of having the names hardcoded here, could we use the static template names from the default template?

Reply:
I could do a `TemplateLookup.itertemplates()`, then test the type so all `_StaticTemplate` would be copied to build directory by default ?